### PR TITLE
Add ConcurrentBag Clear APIs

### PR DESCRIPTION
**Pull request by AlexRadch**
Implement ConcurrentBag.Clear API. See https://github.com/dotnet/corefx/issues/2338 issue.

### CODE REVIEW

**Comment by AlexRadch** [comment]
I do not clear understand  [Adding APIs Guidelines](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/adding-api-guidelines.md) so test project can be compiled and tested locally in `src\System.Collections.Concurrent\tests\` folder but global build return strange errors.

@Priya91 can you help me with config files to add API correctly to pass root build?

File: src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentBag.cs [hunk 1]
```diff
@@ -567,6 +565,33 @@ public T[] ToArray()
...
+            // Short path if the bag is empty
+            if (_headList == null)
+                return;
+
+            bool lockTaken = false;
+            try
+            {
+                FreezeBag(ref lockTaken);
+
+                var currentList = _headList;
```
Comment (AlexRadch) [inline src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentBag.cs]: Changed

**Comment by Priya91** [comment]
@AlexRadch Can you post the build errors..

**Comment by AlexRadch** [comment]
@Priya91 [msbuild.err.txt](https://github.com/dotnet/corefx/files/620205/msbuild.err.txt)

**Comment by stephentoub** [comment]
@AlexRadch, it'll probably be easier for you if you wait for @jusinvp to finish with https://github.com/dotnet/corefx/pull/13828.  As part of that he's added the necessary netcoreapp11 support to the project, and once that's there it should be fairly straightforward for you to add these new methods.

**Comment by AlexRadch** [comment]
@stephentoub Ok. I am waiting when you add netcoreapp11 support to this project. Thanks.

**Comment by karelz** [comment]
@justinvp's PR is merged for 5 days now - https://github.com/dotnet/corefx/pull/13828#event-894330765. The work should be unblocked ...

**Comment by karelz** [comment]
@AlexRadch do you plan to finish it?

**Comment by karelz** [comment]
@AlexRadch ping?

**Comment by stephentoub** [comment]
Thanks for your work on this, @AlexRadch.  Since this has been sitting for a couple of months and needed to be significantly updated based on changes to the target types, I went ahead and took your commits and created a new PR at https://github.com/dotnet/corefx/pull/15292.

**Pull request closed without merging**